### ISSUE TITLE
Use Slim templates instead of ERB

### ISSUE THREAD

**Issue by rdsubhas** [rapidftr/tracker#13]
[Slim](//slim-lang.com/) is a neat and concise templating framework for Ruby. It is miles better than the default ERB templates used in Ruby/Rails. We'd like to use that for our templates since it makes them more readable and easier to maintain.

**Goals:**
- Start porting ERB templates to Slim slowly, beginning with the Login page. We can merge independent pages as and when they are complete.
- Tests should need no changes, since the actual output HTML/CSS should not change. Minor changes for convenience are fine.
- Use two-space indent

We also evaluated HAML and Jade. Slim is more concise than both.

**Comment by larahy** [comment]
I'm happy to take this on, however I have one question: I've been playing around with it a bit and I wonder whether it might be easier to leave the application layout page till last?

**Comment by rdsubhas** [comment]
Hi @larahy We were kinda assuming that every file can be changed independently and Rails would be intelligent enough, but we didn't go deep enough to validate that assumption. So yes, it looks good to leave the application layout till last. Anything that works :)

So should we change the scope of this story to only the Login page to start with? And create separate cards for other pages?

**Comment by sferik** [comment]
I was thinking that this would be a good task for the @RapidRailsGirls team, since they are learning HTML and MVC. Specifically, this would be a good opportunity for them to become familiar with all the views while learning Slim.

@larahy I don’t want to stop you from working on this, especially if you’ve already started. Maybe we can each do part of the job?

@rdsubhas I can confirm that Rails is smart enough to serve different templates with different extensions (e.g. `.html.erb`, `.html.slim`) so it should be possible to make this transition in pieces.

**Comment by sferik** [comment]
> Use two-space indent

This is related to the work already being done in https://github.com/rapidftr/RapidFTR/pull/655.

**Comment by rdsubhas** [comment]
@sferik yep I think there's a good amount of scope for the @RapidRailsGirls team to also work in parallel. We're overall looking at this: Port to Slim :arrow_right: Style with Foundation :arrow_right: Refactor Views. When we port to Foundation, there will certainly be some refactoring of views since much of the present complex HTML/CSS would be made easy and irrelevant with foundation.

So @sferik what do you suggest as a general approach? Is it good to first port all HTML templates to Slim, and then do the styling/refactoring? Or take one chunk of functionality (say the Login page) and do everything (Slim+Foundation) as a bunch?

**Comment by sferik** [comment]
I think it makes the most sense to start by porting to Slim. @kaeff is giving a short lesson on Slim now.

**Comment by rdsubhas** [comment]
Cool :)

@larahy I've changed this story to target only the Login Page.
@sferik I've created a separate story for the Layout - #130

And we'll create more stories as we go, since this has a good amount of scope, and some pages can be really complicated to port.

**Comment by sferik** [comment]
@larahy: We’ve successfully [converted the application layout to Slim](https://github.com/rapidftr/RapidFTR/pull/659). We’re happy to work on this ticket to convert the login page as well, unless you’ve already made progress on this. If we don’t hear from you by the end of the day, we may start working on this tomorrow.

**Comment by larahy** [comment]
@sferik thanks so much, and sorry for the silence from me, was on leave last week but am now back in the office and have just finished doing the login page so just need to work out how to push a finished story ;)

**Comment by sferik** [comment]
@larahy Good to hear!

@rdsubhas It looks like there are currently 100 ERB templates remaining.

```
$ find app/views | grep '.erb$' | wc -l
     100
```

Do we want to convert the rest of the ERB templates to Slim (beyond the application layout and login pages)? If so, this is something I think the @RapidRailsGirls team could work on.

**Comment by rdsubhas** [comment]
@larahy in case you need some pointers, pls check https://github.com/rapidftr/tracker/#implement

**Comment by rdsubhas** [comment]
@sferik @larahy Yep we'd love to create more stories for this and nail it down :) I've created two other stories - [Children pages](rapidftr/tracker#140), [Advanced Search pages](rapidftr/tracker#141)

**Comment by larahy** [comment]
@rdsubhas Thank you, am having a nightmare merging conflicts, think I'm being very stupid! Should have a pull request for you shortly hopefully though and then I can pick up one of the stories you've just created :)

**Comment by rdsubhas** [comment]
@larahy welcome, we'll be in IRC chatroom during the day, http://webchat.freenode.net/?channels=rapidftr if you have any troubles please feel free to ping us there...

**Comment by rdsubhas** [comment]
@sferik @kat-crumbs @BriocheBerlin @RapidRailsGirls @larahy Thanks for taking this forward! At any point, if you think that the ERB -> Slim conversion work is boring or repetitive, please let me know, we can _mixin_ the foundation porting as well, so that when you're doing the Slim you can also use some cool & simpler markup...
